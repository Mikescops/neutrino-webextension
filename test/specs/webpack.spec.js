const path = require('path')
const _ = require('lodash')
const Neutrino = require('neutrino/Neutrino')
const react = require('@neutrinojs/react')
const webext = require('../../lib')

describe('webpack', () => {
  const NODE_ENV = process.env.NODE_ENV
  afterAll(() => {
    process.env.NODE_ENV = NODE_ENV
  })

  describe('development', () => {
    beforeEach(() => {
      process.env.NODE_ENV = 'development'
    })

    describe('snapshots', () => {
      test('entries without polyfill', () => {
        testEntriesSnapshot()
      })

      test('entries with polyfill', () => {
        testEntriesSnapshot(true)
      })
    })
  })

  describe('production', () => {
    beforeEach(() => {
      process.env.NODE_ENV = 'production'
    })

    describe('snapshots', () => {
      test('entries without polyfill', () => {
        testEntriesSnapshot()
      })

      test('entries with polyfill(should also not be injected)', () => {
        testEntriesSnapshot(true)
      })
    })
  })
})

function testEntriesSnapshot (polyfill) {
  const neutrino = new Neutrino(optionsFixture())
  neutrino.use(react())
  neutrino.use(polyfill ? webext({ polyfill: true }) : webext())
  expect(
    _.mapValues(neutrino.config.entryPoints.entries(), entry => {
      return entry.values()
    })
  ).toMatchSnapshot()
}

function optionsFixture () {
  return {
    source: path.resolve(__dirname, '../src'),
    mains: {
      background: {
        entry: 'background',
        webext: {
          type: 'background'
        }
      },
      popup: {
        entry: 'popup',
        webext: {
          type: 'browser_action'
        }
      },
      content1: {
        entry: 'content1',
        webext: {
          type: 'content_scripts',
          setup: 'content1/__dev__/setup',
          manifest: {
            matches: ['<all_urls>']
          }
        }
      },
      content2: {
        entry: 'content2',
        webext: {
          type: 'content_scripts',
          manifest: {
            matches: ['https://github.com/crimx/neutrino-webextension'],
            run_at: 'document_start',
            match_about_blank: true,
            all_frames: true
          }
        }
      }
    }
  }
}
