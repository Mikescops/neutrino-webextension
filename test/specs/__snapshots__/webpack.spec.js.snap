// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack development should add global and entry setup files 1`] = `
Array [
  "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
  "/global-setup-path",
  "/entry-setup-path",
  "/project/src/index",
]
`;

exports[`webpack development should resolve relative setup path 1`] = `
Array [
  "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
  "/project/global-setup-path",
  "/project/src/entry-setup-path",
  "/project/src/index",
]
`;

exports[`webpack development snapshots entries with polyfill 1`] = `
Object {
  "background": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/background.js",
    "/project/src/background",
  ],
  "content1": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/content1/__dev__/setup",
    "/project/src/content1",
  ],
  "content2": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/content2",
  ],
  "page": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/page",
  ],
  "pageless": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/pageless",
  ],
  "popup": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/popup",
  ],
}
`;

exports[`webpack development snapshots entries without polyfill 1`] = `
Object {
  "background": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/background.js",
    "/project/src/background",
  ],
  "content1": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/content1/__dev__/setup",
    "/project/src/content1",
  ],
  "content2": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/content2",
  ],
  "page": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/page",
  ],
  "pageless": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/pageless",
  ],
  "popup": Array [
    "/home/crimx/code/neutrino-webextension/node_modules/webextensions-emulator/dist/core.js",
    "/project/src/popup",
  ],
}
`;

exports[`webpack production snapshots entries with polyfill(should also not be injected) 1`] = `
Object {
  "background": Array [
    "/project/src/background",
  ],
  "content1": Array [
    "/project/src/content1",
  ],
  "content2": Array [
    "/project/src/content2",
  ],
  "page": Array [
    "/project/src/page",
  ],
  "pageless": Array [
    "/project/src/pageless",
  ],
  "popup": Array [
    "/project/src/popup",
  ],
}
`;

exports[`webpack production snapshots entries without polyfill(should not be injected) 1`] = `
Object {
  "background": Array [
    "/project/src/background",
  ],
  "content1": Array [
    "/project/src/content1",
  ],
  "content2": Array [
    "/project/src/content2",
  ],
  "page": Array [
    "/project/src/page",
  ],
  "pageless": Array [
    "/project/src/pageless",
  ],
  "popup": Array [
    "/project/src/popup",
  ],
}
`;
